### PR TITLE
uc8151: add FillRectangle() and SetScroll() functions to satisfy tinyterm.Displayer interface

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds to the `uc8151` display by implementing the `FillRectangle()` and `SetScroll()` functions in order to satisfy the `tinyterm.Displayer` interface.

### CODE REVIEW

**Comment by deadprogram** [comment]
Once this can get merged, I have some related changes to tinyterm that allows it to work on the badger2040 display.

**Review by ydnar — APPROVED** [review]
Do you want to add an assertion that it satisfies the interface?

**Comment by deadprogram** [comment]
@ydnar what would be the best way to do that? The interface is defined in package that consumes this package here: https://github.com/tinygo-org/tinyterm/blob/release/tinyterm.go#L27

**Comment by ydnar** [comment]
Ah. Nevermind then!

**Comment by ydnar** [comment]
> @ydnar what would be the best way to do that? The interface is defined in package that consumes this package here: https://github.com/tinygo-org/tinyterm/blob/release/tinyterm.go#L27

Can you define it elsewhere and have tinyterm import that and alias it?

**Comment by ydnar** [comment]
Seems like a big lift for little value...

**Comment by deadprogram** [comment]
I'll think about a way to do it in a future PR. Thanks for review @ydnar now merging.

File: uc8151/uc8151.go [hunk 1]
```diff
@@ -478,3 +478,31 @@ func (d *Device) SetLUT(speed Speed, flickerFree bool) error {
...
+		}
+	}
+
+	return nil
+}
+
+// SetScroll sets the vertical scrolling for the display, which is a NOP for this display.
+func (d *Device) SetScroll(line int16) {
+	return
+}
```
Comment (aykevl) [inline uc8151/uc8151.go:508]: You have a program that calls `SetScroll` without first calling `SetScrollArea`? I think that won't work on other displays like ST7789.

Comment (deadprogram) [reply uc8151/uc8151.go:508]: This is for interface compatibility with tinyterm which does not call `SetScrollArea()`. See https://github.com/tinygo-org/tinyterm/blob/release/tinyterm.go#L27-L31

Actually, in the case of the `uc8151` tinyterm only uses the software scroll, which is why the implementation here just returns without doing anything.

**Pull request merged**